src/Feature/Search/Feature_Search.re

```diff
@@ -85,10 +85,14 @@ type outmsg =
       filePath: string,
       location: CharacterPosition.t,
     })
+  | PreviewFile({
+      filePath: string,
+      location: CharacterPosition.t,
+    })
   | Focus
   | UnhandledWindowMovement(Component_VimWindows.outmsg);
 
-let update = (model, msg) => {
+let update = (~previewEnabled, model, msg) => {
   switch (msg) {
```

Comment:
One thing I'm thinking about - we could remove injecting this parameter, and instead update the `OpenFile` outmsg to something like:
```
OpenFile({
   filePath: string,
   location: CharacterPosition.t,
   preferToPreview: bool,  // Let the consumer know that, if the configuration allows for previewing, we should preview!
})
```

This way - the feature can communicate the intent - that we should try to preview - without needing to inject additional parameters. Then, when the `OpenFile` outmsg is handled in `Features` - we could decide whether it should be an `openFile` effect or `previewFile` effect based on that property and the user's configuration.